variables/util.go

```diff
@@ -16,6 +16,38 @@ func ParseKeyValue(env string) (string, string) {
 	return parts[0], v
 }
 
+// ParseEscapedKeyValue pases a key-value type into its parts
+// keys and values containing '=', and '\' must be escaped using '\=', and '\\' respectively
+func ParseEscapedKeyValue(s string) (string, string, bool) {
```

Comment:
I wonder if it would be safe to simply replace `ParseKeyValue` with this new method, but I wonder if it'll break any backwards compatibility issues where some a value might contain an `=` which would have to be escaped.